### PR TITLE
Refresh JWKS if no suitable key found in current set

### PR DESCRIPTION
**Pull request by mtrojanowski**

### CODE REVIEW

File: flask_of_oil/jwt_validator.py [hunk 1]
```diff
@@ -81,6 +83,21 @@ def validate(self, jwt):
         # noinspection PyBroadException
         try:
             jws.verify_compact(jwt, self.jwks)
+        except NoSuitableSigningKeys as e:
+            now = datetime.now().timestamp()
+            if self.lastSuccessfulJWKSFetch + self.jwksFetchCooldown < now:
```
Comment (anestos) [inline flask_of_oil/jwt_validator.py:88]: `None + 30` will cause an exception, no?

File: flask_of_oil/jwt_validator.py [hunk 2]
```diff
@@ -133,4 +150,5 @@ def load_keys(self):
         # load the jwk set.
         jwks = KEYS()
         jwks.load_jwks(self.get_jwks_data())
+        self.lastSuccessfulJWKSFetch = datetime.now().timestamp()
```
Comment (anestos) [inline flask_of_oil/jwt_validator.py:153]: `AttributeError: 'datetime.datetime' object has no attribute 'timestamp'`
are you sure that works?

Comment (anestos) [reply flask_of_oil/jwt_validator.py:153]: nvm that was python2. works in python3

File: flask_of_oil/jwt_validator.py [hunk 3]
```diff
@@ -33,13 +33,15 @@ class JwtValidatorException(Exception):
 
 class JwtValidator:
     def __init__(self, jwks_url, issuer, audience, verify_ssl_server=True):
+        self.logger = logging.getLogger(__name__)
         self.supported_algorithms = ['RS256', "RS512"]
         self.jwks_url = jwks_url
         self.aud = audience
         self.iss = issuer
         self.verify_ssl_server = verify_ssl_server
+        self.lastSuccessfulJWKSFetch = None
```
Comment (anestos) [inline flask_of_oil/jwt_validator.py]: should prob initialize to `0` to not have the possibility for the exception mentioned below

**Review by anestos — APPROVED** [review]

File: flask_of_oil/jwt_validator.py (hunk 1 above)
Comment (mtrojanowski) [reply flask_of_oil/jwt_validator.py:88]: Good catch. I was thinking in PHP/JS terms...

**Pull request merged**